File: backend/src/models/Diet.js
const mongoose = require('mongoose');

const mealCheckInSchema = new mongoose.Schema({
  mealId: { type: String, required: true },
  status: { type: String, enum: ['seguiu', 'ajustou', 'nao'], required: true },
  note: { type: String, default: '' },
}, { _id: false });

const freeMealSchema = new mongoose.Schema({
  day: { type: String, default: '' },
  description: { type: String, default: '' },
}, { _id: false });

const cheatEntrySchema = new mongoose.Schema({
  description: { type: String, default: '' },
}, { _id: false });

const dietLogSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  date: { type: Date, required: true },
  checkIns: [mealCheckInSchema],
  freeMeals: [freeMealSchema],
  cheatEntries: [cheatEntrySchema],
}, { timestamps: true });

dietLogSchema.index({ date: -1 });
dietLogSchema.index({ userId: 1 });

module.exports = mongoose.model('DietLog', dietLogSchema);

File: lib/core/models/diet.dart
class DietMeal {
  final String id;
  final String name;
  final String time;
  final String description;
  final int kcal;

  const DietMeal({
    required this.id,
    required this.name,
    required this.time,
    required this.description,
    required this.kcal,
  });
}

enum MealStatus { seguiu, ajustou, nao }

class MealCheckIn {
  final String mealId;
  final MealStatus status;
  final String? note;

  const MealCheckIn({required this.mealId, required this.status, this.note});

  MealCheckIn copyWith({MealStatus? status, String? note}) {
    return MealCheckIn(
      mealId: mealId,
      status: status ?? this.status,
      note: note ?? this.note,
    );
  }
}
